main.py

```diff
@@ -1,19 +1,55 @@
-#!/usr/bin/env python
-# -*- coding: utf-8 -*-
-
-"""
-Challenge:
-
-#. Implement ``Product`` class
-#. Each ``Product`` instance should implement properties:
-    * ``name`` - a product's name, like apple, cheese etc.
-    * ``price`` - a price for a single unit
-#. ``Product`` instance should have a behavior of calculating total
-   price for a specified quantity of goods
-#. Implement ``ShoppingCart`` class
-#. ``ShoppingCart`` instance should combine products instances and
-    corresponding purchased quantities.
-#. ``ShoppingCart`` instance should implement a method to calculate
-   the total price of entire cart.
-
-"""
+class Product:
+    """Takes the name and price of products."""
+
+    def __init__(self, name: str, price: float) -> None:
+        self.name = name
+        self.price = price
+
+    def __repr__(self) -> str:
+        return f'Product "{self.name.capitalize()}" with price {self.price}'
+
+    def __str__(self) -> str:
+        return self.name
+
+    def cost_calc(self, quantity: int | float) -> float:
+        return round(self.price * quantity, 2)
+
+
+class ShoppingCart:
+    """Takes the quantity, adds, displays and counts the total price of product."""
+
+    def __init__(self) -> None:
+        self.list_of_products = []
+        self.list_of_quantities = []
+
+    def __repr__(self) -> str:
+        return '<ShoppingCart>'
+
+    def add_to_list(self, prod: Product, quantity: int | float) -> None:
```

Comment:
```suggestion
    def add_to_list(self, prod: Product, quantity: Union[int, float]) -> None:
```
Ця фішка для версії python^=3.10
Роки через 1.5 - 2 буде поширено, але поки що буде викликати багато проблем зі зворотньою сумістністю.

Reply:
Не помилка, просто для інформації